### PR TITLE
Various config changes to meet the new specification.

### PR DESCRIPTION
**Pull request by Mark104**
To merge working config changes that meet the specification for OAuth / WS as used by server4 / server5.

### CODE REVIEW

**Comment by p2004a** [comment]
Can we already use discovery endpoint and not hardcode the token and authorization endpoints? I see OAuth2 client library supports it: https://github.com/badgateway/oauth2-client/blob/main/src/client.ts#L126

**Comment by Mark104** [comment]
56485c0 changes the OAuth config to use disocvery endpoint rather than specific endpoints as demonstrated by Mongy.

File: src/index.ts [hunk 1]
```diff
@@ -56,11 +56,10 @@ export class TachyonClient<OriginActor extends TachyonActor> {
         this.config = { ...defaultTachyonClientOptions, ...config };
 
         this.oauthClient = new OAuth2Client({
-            server: `http${this.config.ssl ? "s" : ""}://${this.getServerBaseUrl()}`, // TODO: https, discovery, allow specifying custom address,
-            clientId: config.clientId ?? "tachyon_client",
-            clientSecret: config.clientSecret,
-            authorizationEndpoint: "/authorize",
-            tokenEndpoint: "/token",
+            server: `http${this.config.ssl ? "s" : ""}://${this.getServerBaseUrl()}`,
+            clientId: "generic_lobby",
```
Comment (p2004a) [inline src/index.ts]: Sorry missed before, I would keep allowing to override the default value of `client_id` and fix the comment above in line 26. Also config's `client_secret` is no longer used.

Comment (Mark104) [reply src/index.ts]: Addressed in 5964f4

**Pull request merged**